File: part11/11.1/BookAuthor.java
// BookAuthor.java
import java.util.*;

public class BookAuthor{
	public static void main(String[] args) {
		List<Book> books = new ArrayList<>();
		Person james = new Person("James",30);
		Person john = new Person("John",35);

		Book javaProgramming = new Book("Java Programming","John Publishing",james);
		Book jsProgramming = new Book("JavaScript Programming","Acme Publishing",james);
		Book postgresDb = new Book("Postgres","CCP Publishing",john);
		books.add(javaProgramming);
		books.add(jsProgramming);
		books.add(postgresDb);

		books.stream()
			.filter(b -> b.getAuthor().getName().equals("James"))
			.forEach(b -> System.out.println(b));
	}
}

File: part11/11.1/bookAuthor.js
// bookAuthor.js

const books = [];

function book (title,publisher,author){
	return {
		title,
		publisher,
		author
	}
}

const person = (name,age)=> {
	return {
		name,
		age
	}
}

const john = person("John",22);
const jasmine = person("Jasmine",30);

const jsbook = book("Js Programming","Acme Publishing",john);
const javabook = book("Java Programming","James Publishing",john);
const postgres = book("Postgres Db","CCP Publishing",jasmine)

books.push(jsbook);
books.push(javabook);
books.push(postgres);

const booksOfJohn = books.filter((book)=> book.author.name == "John")
console.log(booksOfJohn)